client/src/controllers/TooltipController.ts

```diff
@@ -0,0 +1,56 @@
+import { Controller } from '@hotwired/stimulus';
+import tippy, { Placement, Props, Instance } from 'tippy.js';
+import { hideTooltipOnEsc } from '../includes/initTooltips';
+
+/**
+ * A Tippy.js tooltip with simple popover content.
+ *
+ * @example
+ * <button type="button" data-controller="w-tooltip" data-w-tooltip-content-value="More detail here">
+ *  A button with a tooltip
+ * </button>
+ */
+export class TooltipController extends Controller<HTMLElement> {
+  static values = {
+    content: String,
+    placement: { default: 'bottom', type: String },
+  };
+
+  declare contentValue: string;
+  declare placementValue: Placement;
+  tippy?: Instance<Props>;
+
+  connect() {
+    this.tippy = tippy(this.element, this.options);
+  }
+
+  contentValueChanged(newValue: string, oldValue: string) {
+    if (!oldValue || oldValue === newValue) return;
+    this.tippy?.setProps(this.options);
+  }
+
+  placementValueChanged(newValue: string, oldValue: string) {
+    if (!oldValue || oldValue === newValue) return;
+    this.tippy?.setProps(this.options);
+  }
+
+  hide() {
+    this.tippy?.hide();
+  }
+
+  show() {
+    this.tippy?.show();
+  }
+
+  get options(): Partial<Props> {
+    return {
+      content: this.contentValue,
+      placement: this.placementValue,
+      plugins: [hideTooltipOnEsc],
+    };
+  }
+
+  disconnect(): void {
```

Comment:
Should be no problem to rely on inference of the return type here like we do for other functions above.

```suggestion
  disconnect() {
```

Reply:
Thanks. I think vscode adds this automatically. I forget to remove it sometimes.